### PR TITLE
feat(model): plot

### PR DESCRIPTION
**Pull request by felipehuerta17**
Equivalent to the [Utah Teapot](https://en.wikipedia.org/wiki/Utah_teapot)

`model.plot()` generates a plot from the calculated solution with the most relevant quantities for each of our models:

- SIR: (t vs I and R)
image

- SIR-X: (t vs I and X)
image

Calling `model.plot` requires the model to be either solved or fitted (because fitted -> solved).

Tests & updated README and docs will come soon. This will create a user friendly way for our users to validate the correct behaviour of open-sir against their parameters & fitting.

### CODE REVIEW

**Comment by felipehuerta17** [comment]
WEB API @sasalatart: instead of running the code by console, create a web request that provides the results. Create the API in Go, that will work as a web interface.

An option for the output of plot should be SVG that can be read from Go

File: doc/tutorials/SIR-X.ipynb [hunk 1]
```diff
@@ -501,12 +501,32 @@
...
-   "source": []
+   "source": [
+    "g_sirx.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Looking at the values of the number of infected and quarantined individuals, as well as the time of the peak, it is clear that the both plots are in good agreement, which validates the customized plot created by the code cell below \"long term prediction\"."
```
Comment (sasalatart) [inline doc/tutorials/SIR-X.ipynb:528]:
```suggestion
    "Looking at the values of the number of infected and quarantined individuals, as well as the time of the peak, it is clear that both plots are in good agreement, which validates the customized plot created by the code cell below \"long term prediction\"."
```

File: opensir/models/sir.py [hunk 2]
```diff
@@ -144,6 +145,35 @@ def set_initial_conds(self, array=None, n_S0=None, n_I0=None, n_R0=None):
...
+        fig, ax1 = plt.subplots(figsize=[5, 5])
+
+        color = "tab:red"
+        ax1.set_xlabel("Time / days", size=14)
+        ax1.set_ylabel("Number of infected", size=14)
+        ax1.plot(t, n_i, color=color)
+        ax1.tick_params(axis="y", labelcolor=color)
+        ax1.tick_params(axis="both", labelsize=13)
+
+        ax2 = ax1.twinx()  # instantiate a second axes that shares the x coordinate
```
Comment (sasalatart) [inline opensir/models/sir.py:166]:
```suggestion
        ax2 = ax1.twinx()  # instantiate a second axis that shares the x coordinate
```

File: opensir/models/sirx.py [hunk 3]
```diff
@@ -172,6 +173,36 @@ def set_initial_conds(self, array=None, n_S0=None, n_I0=None, n_R0=None, n_X0=No
...
+        fig, ax1 = plt.subplots(figsize=[5, 5])
+
+        color = "tab:red"
+        ax1.set_xlabel("Time / days", size=14)
+        ax1.set_ylabel("Number of infected $N_I$", size=14)
+        ax1.plot(t, n_i, color=color)
+        ax1.tick_params(axis="y", labelcolor=color)
+        ax1.tick_params(axis="both", labelsize=13)
+
+        ax2 = ax1.twinx()  # instantiate a second axes that shares the x coordinate
```
Comment (sasalatart) [inline opensir/models/sirx.py:195]:
```suggestion
        ax2 = ax1.twinx()  # instantiate a second axis that shares the x coordinate
```

File: opensir/models/sirx.py [hunk 4]
```diff
@@ -172,6 +173,36 @@ def set_initial_conds(self, array=None, n_S0=None, n_I0=None, n_R0=None, n_X0=No
...
+
+        color = "tab:blue"
+        ax2.set_ylabel("Number of quarantined $N_X$", size=14)
+        ax2.plot(t, n_x, color=color, linestyle=":", linewidth=4)
+        ax2.tick_params(axis="y", labelcolor=color)
+        ax2.tick_params(axis="both", labelsize=13)
+
+        plt.title("SIR-X model predictions", size=16)
+        plt.show()
```
Comment (sasalatart) [inline opensir/models/sirx.py:204]: I see there is a lot of repetition between this model and the SIR model. Maybe (not in this PR, but in a future one) we could refactor this to DRY out the code. Maybe use inheritance, or some other pattern.

Comment (felipehuerta17) [reply opensir/models/sirx.py:204]: Let's implement this in another PR for plot, which will also deal with exporting a svg and the cli, following your issue #81

**Pull request merged**